src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -631,7 +649,8 @@ void StringLiteralEntry::DeleteEntry (StringLiteralEntry *pEntry)
     }
     CONTRACTL_END;
 
-    _ASSERTE (VolatileLoad(&pEntry->m_dwRefCount) == 0);
+    _ASSERTE (pEntry->GetRefCount() == 0);
+    _ASSERTE (!pEntry->IsStringFrozen());
```

Comment:
Unfortunately, this assert is not valid. We can come here with frozen string when we fail to add the string to the hashtable here: https://github.com/dotnet/runtime/pull/49576/files#diff-d86558f39cd09fa14c22d6e7ec315bb6dcd717784d90a69680508862ea7d893cR506

Reply:
Fixed, but I am not sure I understand at what conditions it may fail other than OOM https://github.com/dotnet/runtime/blob/8fe12bc5a66a5ef42200a3d88d68b5991eb158e6/src/coreclr/vm/eehash.inl#L228-L278 - it doesn't look like it fails to insert if the key is already presented, unless I am looking at a wrong impl.